src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_data_item_schema.cc

```diff
@@ -9,33 +9,78 @@ CREATE_LOGGERPTR_LOCAL(vehicle_data_logger, "VehicleDataItemSchema");
+          policy_data_item.params->end(),
+          std::back_inserter(full_history_vector),
+          [&name](VehicleDataItem& item) { return item.name == name; });
+
+      std::sort(full_history_vector.begin(),
+                full_history_vector.end(),
+                [](const policy_table::VehicleDataItem& left,
+                   const policy_table::VehicleDataItem& right) {
+                  if (!right.since.is_initialized()) {
+                    return false;
+                  }
+                  if (!left.since.is_initialized()) {
+                    return true;
+                  }
+                  const std::string l = *left.since;
+                  const std::string r = *right.since;
+                  return std::stof(l.c_str()) > std::stof(r.c_str());
+                });
+
+      std::vector<smart_objects::SMember> history_vector = {};
+      for (auto history_iterator = full_history_vector.begin() + 1;
+           history_iterator < full_history_vector.end();
+           ++history_iterator) {
+        auto schema_item = PolicyDataItem(*history_iterator);
+        smart_objects::SMember member(create(schema_item, schema_type),
+                                      bool(history_iterator->mandatory),
+                                      std::string(*(history_iterator->since)),
+                                      std::string(*(history_iterator->until)),
+                                      bool(*(history_iterator->deprecated)),
+                                      bool(*(history_iterator->removed)));
+        history_vector.push_back(member);
+      }
+
+      const auto vehicle_data_item = full_history_vector.begin();
+      auto schema_item = PolicyDataItem(*vehicle_data_item);
       smart_objects::SMember member(create(schema_item, schema_type),
-                                    bool(e.mandatory),
-                                    std::string(*e.since),
-                                    std::string(*e.until),
-                                    bool(*e.deprecated),
-                                    bool(*e.removed));
-      std::string name;
+                                    bool(vehicle_data_item->mandatory),
+                                    std::string(*(vehicle_data_item->since)),
+                                    std::string(*(vehicle_data_item->until)),
+                                    bool(*(vehicle_data_item->deprecated)),
+                                    bool(*(vehicle_data_item->removed)),
+                                    history_vector);
+
+      std::string member_name;
       switch (schema_type) {
         case VehicleDataItemSchema::SchemaType::HMI:
-          name = e.key;
+          member_name = vehicle_data_item->key;
           break;
         case VehicleDataItemSchema::SchemaType::MOBILE:
         default:
-          name = e.name;
+          member_name = vehicle_data_item->name;
           break;
       }
```

Comment:
@mkorniichuk I propose to use ternary operator instead of `switch` statement:
```suggestion
      std::string member_name = schema_type == VehicleDataItemSchema::SchemaType::HMI ?
      vehicle_data_item->key : vehicle_data_item->name;
```